core/domain/user_id_migration.py

```diff
@@ -0,0 +1,515 @@
+
+    @staticmethod
+    def map(user_model):
+        """Implements the map function for this job."""
+        gae_id = user_model.gae_id
+        if (len(user_model.id) != 32 or
+                not all(c.islower() for c in user_model.id)):
+            yield ('FAILURE - WRONG ID FORMAT', (gae_id, user_model.id))
+        success = True
+        for model_class in models.Registry.get_all_storage_model_classes():
+            base_classes = [
+                base.__name__ for base in inspect.getmro(model_class)]
+            # BaseSnapshotMetadataModel and models that inherit from it
+            # are checked from the associated VersionedModel.
+            if 'BaseSnapshotMetadataModel' in base_classes:
+                continue
+            # BaseSnapshotContentModel and models that inherit from it
+            # are checked from the associated VersionedModel.
+            if 'BaseSnapshotContentModel' in base_classes:
+                continue
+            if (model_class.get_deletion_policy() ==
+                    base_models.DELETION_POLICY.NOT_APPLICABLE):
+                continue
+            if model_class.has_reference_to_user_id(gae_id):
+                yield ('FAILURE - HAS REFERENCE TO GAE ID',
+                       (gae_id, model_class.__name__))
+                success = False
+
+        if success:
+            yield ('SUCCESS', (gae_id, user_model.id))
+
+    @staticmethod
+    def reduce(key, status):
+        """Implements the reduce function for this job."""
+        yield (key, status)
+
+
+class ModelsUserIdsHaveUserSettingsVerificationJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for going through all the models that contain user IDs. This
+    job checks that all the user IDs used in the model have their corresponding
+    UserSettingsModel defined.
+    """
+
+    @staticmethod
+    def _does_user_settings_model_exits(user_id):
+        """Check if UserSettingsModel exists for the user_id.
+
+        Args:
+            user_id: str. User ID that should have its UserSettingsModel.
+
+        Returns:
+            True if UserSettingsModel with id equal to user_id exists, False
+            otherwise.
+        """
+        if user_id == feconf.SYSTEM_COMMITTER_ID:
+            return True
+        return user_models.UserSettingsModel.get_by_id(user_id) is not None
+
+    @staticmethod
+    def _check_id_and_user_id_exist(model_id, user_id):
+        """Check if UserSettingsModel exists for user_id and model id contains
+        user_id.
+
+        Args:
+            model_id: str. ID of the model that should contain the user_id.
+            user_id: str. User ID that should have its UserSettingsModel.
+
+        Returns:
+            True if UserSettingsModel with id as user_id in model exists, False
+            otherwise.
+        """
+        if user_id not in model_id:
+            return False
+        if user_id == feconf.SYSTEM_COMMITTER_ID:
```

Comment:
Why can't this be absorbed in the get_by_id "return" below?

Reply:
We don't have UserSettingsModel for `SYSTEM_COMMITTER_ID`, thus we have to have if for that.

Reply:
Then the docstring seems incorrect. Could you please elaborate on this special case? Ditto below.

Reply:
Updated, done.